experimental/list_formatter/src/patterns.rs

```diff
@@ -0,0 +1,129 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::list_formatter::{Type, Width};
+
+// This should be a compact representation of the CLDR data. Each locale entry is a 3 x 3 x 4 array 
+// (type x width x 4 patterns) of &str. These refs use 592 bytes of memory on a 64-bit platform,
+// plus the memory actually required by the strings. As there aren't many unique patterns, this is
+// probably negligible (see the test below).
+type RawPatterns<'a> = [[[&'a str;4];3];3];
+const RAW_PATTERNS: &[(&'static str, RawPatterns<'static>)] = &[
+    ("en", [
+        [
+            ["{0}, {1}", "{0} and {1}", "{0}, {1}", "{0}, and {1}"], 
+            ["{0}, {1}", "{0} and {1}", "{0}, {1}", "{0}, {1}"], 
+            ["{0}, {1}", "{0} & {1}", "{0}, {1}", "{0}, & {1}"]
+        ],
+        [
+            ["{0}, {1}", "{0} or {1}", "{0}, {1}", "{0}, or {1}"], 
+            ["{0}, {1}", "{0} or {1}", "{0}, {1}", ", or "], 
+            ["{0}, {1}", "{0} or {1}", "{0}, {1}", "{1}, or {1}"]
+        ],
+        [
+            ["{0}, {1}", "{0}, {1}", "{0}, {1}", "{0}, {1}"], 
+            ["{0}, {1}", "{0}, {1}", "{0}, {1}", "{0}, {1}"], 
+            ["{0}, {1}", "{0}, {1}", "{0}, {1}", "{0}, {1}"]
```

Comment:
Suggestion: In cases like this, you can perform "manual" horizontal inheritance; i.e., the first pattern is required, and the second two are `Option`al, and if missing, then you fall back to the first one.

Reply:
Let's try to solve this in the data layer and only change it here if that doesn't work.

This specific "data implementation" does horizontal inlining by deduping static strings.